### ISSUE TITLE
Upside down video calls

### ISSUE THREAD

**Issue by buggymonkey** [iNPUTmice/Conversations#3833]
#### General information

* **Version:** 2.8.8+fcr
* **Device:** Fairphone 2
* **Android Version:**  LineageOS 17.1 without google things
* **Server name:** self hosted
* **Server software:** ejabberd
* **Conversations source:** F-Droid

#### Steps to reproduce

1. make a video call

#### Expected result

What is the expected output? 
The video of the front cameras should be shown in correct orientation

#### Actual result

What do you see instead?
The video of my front camera is upside down. The video of the other person is oriented correctly. So this might be a Fairphone 2 problem. However, other apps have the front camera oriented correctly.

Thanks for all the effort you spent in this project :heart:

**Comment by m4lvin** [comment]
I have the same problem, also with google-free LineageOS 17.1 (from 24 July) and Conversations 2.8.9+fcr.

The orientation of the front camera is correct in other apps, including the stock camera, OpenCamera, Nextcloud Talk and Firefox.

**Comment by iNPUTmice** [comment]
Please include your device in any 'me too' style bug reports.

**Comment by iNPUTmice** [comment]
Does it show up correctly on the recipient's end?

**Comment by m4lvin** [comment]
> Does it show up correctly on the recipient's end?

No. The video is upside down both on my own FP2 and the person receiving the call (no matter if it is an FP2 or not).

**Comment by iNPUTmice** [comment]
This seems to be device specific and this will probably take someone with
access to the device to fix it for themselves.

The good news is that all webrtc code is wrapped in just one class
https://github.com/iNPUTmice/Conversations/blob/master/src/main/java/eu/siacs/conversations/xmpp/jingle/WebRTCWrapper.java

Someone probably needs to apply some 180 degree 'correction' to the camera
capturer. But only on FP2 devices. And the test all that properly for
rotating the device and for front and back camera.

>

**Comment by m4lvin** [comment]
One more parameter: The FP2 comes with different front camera modules: the original had 2MP, the new module has 5MP [1]. Maybe the problem only occurs with the new one (which I am using). @buggymonkey which front camera module do you have?

I also started a topic here https://forum.fairphone.com/t/front-camera-video-upside-down-in-conversations-video-calls/63110

[1] https://shop.fairphone.com/en/fairphone-2-top-module

**Comment by buggymonkey** [comment]
I also use the new front camera (5MP).

I took a rough look at the WebRTCWrapper file. But i failed to find a proper documentation for the `org.webrtc` stuff. Is there any?

I could also test things on my device.

**Comment by iNPUTmice** [comment]
Reading up on the on Fairphone forum thread it seems that there are Fairphone 2s with and without the new camera that work correctly, meaning a relatively simple `if phone == fairphone2; flip image` condition would even yield false positives.

I’m much inclined to close here since I don’t think there is anything Conversations can do about this. This either seems to be a driver problem or something that could maybe be handled in libwebrtc. On that note; you might want to try the PlayStore version (or compile it yourself) because PlayStore is on a more recent version of libwebrtc. (Not that I have any reason to assume that this is in fact a bug that has been fixed in libwebrtc)

**Comment by buggymonkey** [comment]
I self-compiled Conversations using http://gultsch.de/files/libwebrtc-m84.aar, which doesn't help.
I don't managed to self-compile libwebrtc yet.
@iNPUTmice which versions ob libwebrtc are used in the F-Droid and PlayStore variants?

**Comment by iNPUTmice** [comment]
> I self-compiled Conversations using http://gultsch.de/files/libwebrtc-m84.aar, which doesn't help.

Ok. Thank you for checking.

> I don't managed to self-compile libwebrtc yet.

I don’t think this will be necessary. It's unlikely to produce different results than using the precompiled m84

PlayStore is on m84. F-Droid is on m79

**Comment by LibrEars** [comment]
Hey all,

'me too' (FP2, LineageOS 17.1 without blobbs) Conversations 2.9.2+fcr from F-Droid - I love it!)

Would it be an option to implement a setting to rotate the camera manually (this might cover some other use-cases as well)?

Thank you for all the effort you spent in this project also from my side!

**Comment by qoheniac** [comment]
Here is the corresponding issue on Fairphone's bug tracker:

https://bugtracker.fairphone.com/project/fairphone-android-7/issue/167

**Comment by licaon-kter** [comment]
@qoheniac same for 2.9.4? F-Droid or Play?

**Comment by qoheniac** [comment]
Quicksy 2.9.4 (F-Droid)
Fairphone 2 (5 Mpx)
Fairphone Open 19.12.1 (Android 7.1.2, Linux 3.4.0)

**Comment by wibrt** [comment]
FP2 with Fairphone 21.08.1 (Android 9) has the same problem 
(Conversations from f-droid)

**Comment by qoheniac** [comment]
See also https://issuetracker.fairphone.com/products/fp2-a9/-/issues/21

**Comment by wibrt** [comment]
> See also https://issuetracker.fairphone.com/products/fp2-a9/-/issues/21

info: i don't have this issue with the stock android camera app as described here

**Comment by qoheniac** [comment]
That's because they fixed it there (see Karsten Tausche's reply to the second comment in the linked issue)
